### PR TITLE
add support for XRay Tracing configuration in Lambda CDK Construct

### PR DESCRIPTION
**Pull request by SeekerWing**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

* https://github.com/awslabs/aws-cdk/issues/671
* add support for tracing config
* further reading
  * https://docs.aws.amazon.com/lambda/latest/dg/lambda-x-ray.html
  * https://docs.aws.amazon.com/xray/latest/devguide/xray-services-lambda.html

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/lib/lambda.ts [hunk 1]
```diff
@@ -106,6 +122,12 @@ export interface FunctionProps {
      */
     deadLetterQueue?: sqs.QueueRef;
 
+    /**
+     * Enable AWS X-Ray Tracing for Lambda Function.
+     *
+     * @default undefined X-Ray tracing disabled
+     */
+    tracingMode?: TracingMode;
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda/lib/lambda.ts]: Rename to “tracing”

File: packages/@aws-cdk/aws-lambda/lib/lambda.ts [hunk 2]
```diff
@@ -7,6 +7,22 @@ import { FunctionVersion } from './lambda-version';
 import { cloudformation, FunctionArn, FunctionName } from './lambda.generated';
 import { Runtime } from './runtime';
 
+/**
+ * X-Ray Tracing Modes (https://docs.aws.amazon.com/lambda/latest/dg/API_TracingConfig.html)
+ */
+export enum TracingMode {
+    /**
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda/lib/lambda.ts]: Add “Disabled” option

File: packages/@aws-cdk/aws-lambda/test/test.lambda.ts [hunk 3]
```diff
@@ -911,6 +911,220 @@ export = {
...
+      const stack = new cdk.Stack();
+
+      new lambda.Function(stack, 'MyLambda', {
+          code: new lambda.InlineCode('foo'),
+          handler: 'index.handler',
+          runtime: lambda.Runtime.NodeJS610,
+          tracingMode: lambda.TracingMode.Active
+      });
+
+      expect(stack).toMatch(
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda/test/test.lambda.ts]: Use “expect(stack).to(haveResource(“ so test will be less heavy handed

File: packages/@aws-cdk/aws-lambda/lib/lambda.ts (hunk 1 above)
Comment (SeekerWing) [reply packages/@aws-cdk/aws-lambda/lib/lambda.ts]: done

File: packages/@aws-cdk/aws-lambda/lib/lambda.ts (hunk 2 above)
Comment (SeekerWing) [reply packages/@aws-cdk/aws-lambda/lib/lambda.ts]: done

File: packages/@aws-cdk/aws-lambda/test/test.lambda.ts (hunk 3 above)
Comment (SeekerWing) [reply packages/@aws-cdk/aws-lambda/test/test.lambda.ts]: done

File: CHANGELOG.md [hunk 4]
```diff
@@ -50,6 +50,7 @@
     - AWS::Glue::Classifier XMLClassifier (__added__)
     - AWS::Glue::Crawler Configuration (__added__)
     - AWS::Lambda::Lambda DLQConfigurationSupport (__added__)
+    - AWS::Lambda::Lambda XRayTracingSupport (__added__)
```
Comment (rix0rrr) [inline CHANGELOG.md]: Don't edit this, this is not a list of L2 features, it's a list of L1 changes.

**Pull request merged**